pkg/virt-handler/notify-server/server.go

```diff
@@ -150,13 +150,14 @@ func RunServer(virtShareDir string, stopChan chan struct{}, c chan watch.Event,
 	}()
 
 	// wait for either the server to exit or stopChan to signal
-	select {
-	case <-done:
-		log.Log.Info("notify server done")
-	case <-stopChan:
-		grpcServer.Stop()
-		log.Log.Info("notify server stopped")
+	for {
+		select {
+		case <-done:
+			log.Log.Info("notify server done")
+			return nil
+		case <-stopChan:
+			grpcServer.Stop()
```

Comment:
There is a real problem that the GRPC Stop doesn't monitor the go routines it created from a specific stage onward, and then there will be the panic, after the channel is closed, wrote more info in the issue #3575

i will work on it if its ok please, thanks

Reply:
oh interesting. makes sense.